third_party/intel/include/TritonIntelGPUToLLVM/Utility.h

```diff
@@ -0,0 +1,225 @@
+  return createPredicatedBlock(rewriter, loc, cond, {}, thenOpsFn);
+}
+
+Value storeShared(ConversionPatternRewriter &rewriter, Location loc, Value ptr,
+                  Value val, Value pred);
+
+Value loadShared(ConversionPatternRewriter &rewriter, Location loc, Value ptr,
+                 Type elemTy, Value pred);
+
+Value shuffleXor(Location loc, ConversionPatternRewriter &rewriter, Value val,
+                 int i);
+Value shuffleUp(Location loc, ConversionPatternRewriter &rewriter, Value val,
+                int i);
+Value shuffleIdx(Location loc, ConversionPatternRewriter &rewriter, Value val,
+                 int i);
+Value shuffleIdx(Location loc, ConversionPatternRewriter &rewriter, Value val,
+                 Value i);
+
+Value addStringToModule(Location loc, ConversionPatternRewriter &rewriter,
+                        StringRef key, StringRef content,
+                        unsigned addressSpace);
+
+LLVM::LLVMFuncOp getSpirvPrintfDeclaration(ConversionPatternRewriter &rewriter);
+
+static Value getStackPointer(PatternRewriter &rewriter,
+                             FunctionOpInterface funcOp) {
+  auto mod = funcOp->getParentOfType<ModuleOp>();
+  LLVM::LLVMPointerType ptrTy = ptr_ty(
+      rewriter.getContext(), TritonGEN::TritonGENMemorySpace::kWorkgroup);
+  if (mod->getAttrOfType<IntegerAttr>("triton_gpu.shared").getInt() == 0)
+    return rewriter.create<LLVM::UndefOp>(funcOp.getLoc(), ptrTy);
+  return funcOp.getArgument(funcOp.getNumArguments() - 1);
+}
+
+static Value getSharedMemoryBase(Location loc,
+                                 ConversionPatternRewriter &rewriter,
+                                 Operation *op) {
+  auto ptrTy = LLVM::LLVMPointerType::get(rewriter.getContext(), 3);
+  FunctionOpInterface func =
+      op->template getParentOfType<FunctionOpInterface>();
+  assert(op->hasAttr("allocation.offset"));
+  size_t offset = op->getAttr("allocation.offset")
+                      .cast<IntegerAttr>()
+                      .getValue()
+                      .getZExtValue();
+  Value offVal = i32_val(offset);
+  Value base =
+      gep(ptrTy, i8_ty, LLVM::intel::getStackPointer(rewriter, func), offVal);
+  return base;
+}
+
+// Returns a Value for the format string, which you can reuse.
+Value llPrintf(ConversionPatternRewriter &rewriter, StringRef msg,
+               ValueRange args);
+
+void llPrintf(ConversionPatternRewriter &rewriter, Value msg, ValueRange args);
+
+static Value getModuleWarpSize(RewriterBase &rewriter, Location loc) {
+  auto mod = rewriter.getBlock()->getParent()->getParentOfType<ModuleOp>();
+  return i32_val(triton::gpu::TritonGPUDialect::getThreadsPerWarp(mod));
+}
+
+} // namespace intel
+} // namespace LLVM
+
+using ::mlir::triton::gpu::intel::DpasEncodingAttr;
+
+static SmallVector<Value>
+emitBaseIndexForDpasLayout(Location loc, RewriterBase &rewriter,
+                           const DpasEncodingAttr &dpasLayout,
```

Comment:
```suggestion
                           DpasEncodingAttr dpasLayout,
```

Same for other `Attribute` args